microservices/download/utils.js

```diff
@@ -111,175 +107,193 @@ async function displayError(req, res, errMsg) {
 
 async function findLink(os, type) {
   try {
-    let repositoryQuery = `
-      query getRepositoryBuildStatuses {
-        repository(id: 6483909499158528) {
-          builds(branch: "master", last: 10) {
-            edges {
-              node {
-                id
-                status
-              }
-            }
-          }
+
+    let releases = await doRequest();
+
+    // Now these releases should be sorted already, if we find they aren't we might
+    // have to add semver as a dep on this microservice, which is no fun since this
+    // microservice has 0 deps currently. For now lets assume it's a sorted array
+    // This same assumption is made on the `pulsar-updater` core package
+
+    for (let version of releases) {
+      for (let asset of version.assets) {
+
+        let name = asset?.name;
+
+        let returnObj = {
+          ok: true,
+          content: asset?.browser_download_url
+        };
+
+        // Ensure we have valid data to work with
+        if (typeof name !== "string" && typeof returnObj.content !== "string") {
+          continue;
         }
```

Comment:
Is this meant to be an OR `||` check? Currently an AND `&&`.

<details>

If no `name` OR no `returnObj.content`? Or is "both being missing" the real indicator of bad data?

Is a binary (an `asset`) with _just_ a `name` or _just_ a `browser_download_url` possible/valid? Should lacking _one_ of those cause the loop to `continue` to try the next asset?

</details>

```suggestion
        // Ensure we have valid data to work with
        if (typeof name !== "string" || typeof returnObj.content !== "string") {
          continue;
        }
```

^ I'm assuming this is what is intended? Bail if _either_ `name` or `browser_download_url` is missing, and try the next asset instead?

Reply:
Fantastic catch! You are totally correct, we need to bail if either of these is invalid, glad you saw this!